src/Proxy/AbstractProxy.php

```diff
@@ -204,11 +205,13 @@ protected function prepareArgsLine(ReflectionFunctionAbstract $functionLike)
      */
     protected function getOverriddenFunction(ReflectionFunctionAbstract $functionLike, $body)
     {
-        $reflectionReturnType = PHP_VERSION_ID >= 50700 ? $functionLike->getReturnType() : '';
+        $reflectionReturnType = PHP_VERSION_ID >= 70000 ? $functionLike->getReturnType() : '';
         $modifiersLine        = '';
         if ($reflectionReturnType) {
-            $nsPrefix             = $reflectionReturnType->isBuiltin() ? '' : '\\';
-            $reflectionReturnType = $nsPrefix . (string)$reflectionReturnType;
+            $nullablePrefix = $reflectionReturnType->allowsNull() ? '?' : '';
+            $nsPrefix       = $reflectionReturnType->isBuiltin() ? '' : '\\';
+
+            $reflectionReturnType = $nullablePrefix . $nsPrefix . (string) $reflectionReturnType;
```

Comment:
@smuggli nice catch! In this part of PR everything is ok, because return type can be nullable only in 7.1, for 7.0 return types `allowsNull()` will be false for every case.

But there is a bug in the first part with checking for nullable parameters. So, yes, confirmed as a bug for 7.0.